freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/CloudIdentitySyncService.java

```diff
@@ -0,0 +1,99 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.CloudIdentity;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.service.freeipa.user.model.FmsUser;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UmsUsersState;
+import com.sequenceiq.sdx.api.endpoint.SdxEndpoint;
+import com.sequenceiq.sdx.api.model.SetRangerCloudIdentityMappingRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
+
+@Service
+public class CloudIdentitySyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CloudIdentitySyncService.class);
+
+    @Inject
+    private SdxEndpoint sdxEndpoint;
+
+    public void syncCloudIdentites(Stack stack, UmsUsersState umsUsersState, BiConsumer<String, String> warnings) {
+        if (CloudPlatform.AZURE.equalsIgnoreCase(stack.getCloudPlatform())) {
+            syncAzureObjectIds(stack, umsUsersState, warnings);
+        }
+    }
+
+    private void syncAzureObjectIds(Stack stack, UmsUsersState umsUsersState, BiConsumer<String, String> warnings) {
+        String envCrn = stack.getEnvironmentCrn();
+        LOGGER.info("Syncing Azure Object IDs for environment {}", envCrn);
+
+        Map<String, List<CloudIdentity>> userCloudIdentites = getUserCloudIdentitiesToSync(umsUsersState);
+
+        Map<String, String> userToAzureObjectIdMap = getAzureObjectIdMap(userCloudIdentites);
+        Map<String, String> groupToAzureObjectIdMap = getAzureObjectIdMap(umsUsersState.getGroupToCloudIdentityMap());
+
+        SetRangerCloudIdentityMappingRequest setRangerCloudIdentityMappingRequest = new SetRangerCloudIdentityMappingRequest();
+        setRangerCloudIdentityMappingRequest.setAzureUserMapping(userToAzureObjectIdMap);
+        setRangerCloudIdentityMappingRequest.setAzureGroupMapping(groupToAzureObjectIdMap);
+        try {
+            // TODO The SDX endpoint currently sets the config and triggers refresh. The SDX endpoint should also be updated
+            //      to allow polling the status of the refresh.
+            sdxEndpoint.setRangerCloudIdentityMapping(envCrn, setRangerCloudIdentityMappingRequest);
+        } catch (Exception e) {
+            LOGGER.warn("Failed to set cloud identity mapping for environment {}", envCrn, e);
+            warnings.accept(envCrn, "Failed to set cloud identity mapping");
+        }
+    }
+
+    private Map<String, String> getAzureObjectIdMap(Map<String, List<CloudIdentity>> cloudIdentityMapping) {
```

Comment:
some log here and in `getOptionalAzureObjectId` might be useful to investigate if something is not as desired